lib/tasks/locale.rake

```diff
@@ -302,11 +305,28 @@ namespace :locale do
 
       combined_dir = File.join(Rails.root, "locale/combined")
       Dir.mkdir(combined_dir, 0o700)
-      po_files.each_key do |locale|
+      po_files.each do |locale, files|
+        files.delete_if do |file|
+          puts "Running 'msgfmt --check' on file: #{file}"
+          if system "msgfmt --check #{file}"
+            false
+          else
+            puts "Fatal error running 'msgfmt --check' on file: #{file}.  Review the output above."
+            puts "Continue anyway?(Y/N)"
+            exit 1 if STDIN.gets.strip.upcase == "N"
```

Comment:
@Fryguy do you have suggestions here? Do you know if any tooling runs `po_to_json` as part of the build?  I may have to drive the answer to this question from an ENV variable or just `exit 1` regardless.

I put this here so I can generate the correct json file, excluding the oracle cloud provider's po file since it's invalid, here: https://github.com/ManageIQ/manageiq-ui-classic/pull/8204

I can do followups to remove the pot/po files in all the providers but wanted to get this in first.

Reply:
>> I may have to drive the answer to this question from an ENV variable or just exit 1 regardless.

> I'm thinking just exit 1 regardless. I do like having the check in place. If it's wrong, there's no reason to continue until it's fixed, and the "skip" logic overcomplicates this.

Agreed.  The skip was an exception just to get beyond the oracle provider.

>> I can do followups to remove the pot/po files in all the providers but wanted to get this in first.

> I think we should just drop the oracle file first. It's a straight up bug as is.

Ok, I can drop the pot and initializer there.  That would allow me to easily drop the logic for asking to continue as I can then `exit 1`

Reply:
> Additionally, we don't need the "running msgfmt" comment for each individual file. I could see only once before the loop saying "Running msgfmt --check on all po files". In the good path, all files will succeed and there is no output to stdout anyway. On failure it will write the msgfmt output and then it will say which file you are running msgfmt --check against, so we can easily reproduce.

Good point.  This was nice for debugging but now that the check does the right thing and shows what's failing, it's not needed.